### PR TITLE
docs: Manage Pod resources

### LINKED ISSUE #1015: Pod resource management and scheduling
### Describe the Issue

We need to provide Kubernetes users with guidance for configuring Pods, specifically:

- How to configure the [`resources` object](https://github.com/redpanda-data/helm-charts/blob/e228c875180d5aa6ac3918de45de02cc8b2583d3/charts/redpanda/values.yaml#L146).

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: February 17 

Fixes #1015

I'll backport these changes to 22.3 when this PR is approved.

### CODE REVIEW

File: docs/manage/kubernetes/manage-resources.mdx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+```yaml title="cpu-cores.yaml"
+resources:
+  cpu:
+    cores: <number-of-cpu-cores>
+```
+
+```bash
+helm upgrade --install redpanda redpanda/redpanda -n redpanda --create-namespace \
+  --values cpu-cores.yaml
```
Comment (joejulian) [inline docs/manage/kubernetes/manage-resources.mdx:48]:
```suggestion
  --values cpu-cores.yaml --reuse-values
```

File: docs/manage/kubernetes/manage-resources.mdx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+```yaml title="cpu-cores-overprovisioned.yaml"
+resources:
+  cpu:
+    cores: <number-of-cpu-cores>
+    overprovisioned: true
+```
+
+```bash
+helm upgrade --install redpanda redpanda/redpanda -n redpanda --create-namespace \
+  --values cpu-cores-overprovisioned.yaml
```
Comment (joejulian) [inline docs/manage/kubernetes/manage-resources.mdx:101]:
```suggestion
  --values cpu-cores-overprovisioned.yaml --reuse-values
```

File: docs/manage/kubernetes/manage-resources.mdx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+On a worker node, you can have the following running processes: Kubernetes, Redpanda, and the Seastar subsystem that Redpanda is built on. Each of these processes consumes memory. You can configure the memory resources that are allocated to these processes.
+
+By default, the Helm chart allocates 80% of the configured memory to Redpanda, leaving the rest for the Seastar subsystem and other container processes. Redpanda recommends this default setting.
+
+:::note
+Although you can also allocate the exact amount of memory for Redpanda and the Seastar subsystem manually, Redpanda does not recommend this approach because setting the wrong values can lead to performance issues, instability, or data loss. As a result, this approach is not documented here.
+:::
+
+Redpanda supports single decimal numbers and only the following memory resource units: k M G Ki Mi Gi. For a description of memory resource units, see the [Kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory).
```
Comment (joejulian) [inline docs/manage/kubernetes/manage-resources.mdx:130]: What are "single decimal numbers"?

I also don't think this is true for the helm chart. We convert any SI unit at any precision to the nearest megabyte.

File: docs/manage/kubernetes/manage-resources.mdx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  memory:
+    container:
+      # Omit min to set it to the same value as max.
+      # min:
+      max: <number><unit>
+```
+
+```bash
+helm upgrade --install redpanda redpanda/redpanda -n redpanda --create-namespace \
+  --values memory.yaml
```
Comment (joejulian) [inline docs/manage/kubernetes/manage-resources.mdx:147]:
```suggestion
  --values memory.yaml --reuse-values
```

**Review by shorgi — APPROVED** [review]
See comments, rest LGTM

File: docs/manage/kubernetes/manage-resources.mdx (hunk 3 above)
Comment (JakeSCahill) [reply docs/manage/kubernetes/manage-resources.mdx:130]: this came from the chart: https://github.com/redpanda-data/helm-charts/blob/276201e82d0537e628cc9b5bd5d54613611a48ca/charts/redpanda/values.yaml#L213

Comment (JakeSCahill) [reply docs/manage/kubernetes/manage-resources.mdx:130]: We should rephrase to 'supports one decimal place'

File: docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-best-practices.mdx [hunk 5]
```diff
@@ -149,4 +149,43 @@ To protect your Kubernetes cluster, do the following:
...
-- [Network encryption through TLS](../../../../../manage/kubernetes/security/kubernetes-tls)
\ No newline at end of file
+- [Network encryption through TLS](../../../../../manage/kubernetes/security/kubernetes-tls)
+
+## Set resource requests and limits for memory and CPU
+
+In a production cluster, the resources you allocate to Redpanda should be proportionate to your machine type. Redpanda recommends that you determine and set these values before deploying the cluster, but you can also update the values on a running cluster.
+
+:::info
+You cannot decrease the number of CPU cores in a running Redpanda cluster.
```
Comment (JakeSCahill) [inline docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-best-practices.mdx:159]:
```suggestion
In a running Redpanda cluster, you can only increase the number of CPU cores. You cannot decrease the number of CPU cores in a running cluster.
```

File: docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-best-practices.mdx [hunk 6]
```diff
@@ -149,4 +149,43 @@ To protect your Kubernetes cluster, do the following:
...
+:::
+
+Redpanda recommends that you allocate the following memory and CPU resources:
+
+- At least 4 CPU cores.
+
+- At least 2Gi of memory per core for Redpanda.
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-best-practices.mdx:166]: Users must use Gi in the configuration, so I'm unsure if we should change this.

File: docs/manage/kubernetes/manage-resources.mdx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+</TabItem>
+</Tabs>
+
+:::note
+<ValuesYaml path="resources.cpu"/>
+:::
+
+## Configure memory resources
+
+On a worker node, you can have the following running processes: Kubernetes, Redpanda, and the Seastar subsystem that Redpanda is built on. Each of these processes consumes memory. You can configure the memory resources that are allocated to these processes.
```
Comment (RafalKorepta) [inline docs/manage/kubernetes/manage-resources.mdx:122]: I would be cautious with the statement:
> you can have the following running processes: Kubernetes, Redpanda, and the Seastar subsystem that Redpanda is built on

One might think that Redpanda schedules/runs another process. That's not a case, as Redpanda is single binary.

**Review by RafalKorepta — APPROVED** [review]
One nit that doesn't prevent from merging.

File: docs/manage/kubernetes/manage-resources.mdx (hunk 3 above)
Comment (JakeSCahill) [reply docs/manage/kubernetes/manage-resources.mdx:130]: @joejulian is the comment in the chart incorrect?

File: docs/manage/kubernetes/manage-resources.mdx (hunk 7 above)
Comment (JakeSCahill) [reply docs/manage/kubernetes/manage-resources.mdx:122]: @RafalKorepta would this be better?

On a worker node, Kubernetes processes and Redpanda processes are running at the same time, including the Seastar subsystem that is built into the Redpanda binary. Each of these processes consumes memory. You can configure the memory resources that are allocated to these processes.

Comment (RafalKorepta) [reply docs/manage/kubernetes/manage-resources.mdx:122]: Perfect 👍

Comment (JakeSCahill) [inline docs/manage/kubernetes/manage-resources.mdx:122]:
```suggestion
On a worker node, Kubernetes processes and Redpanda processes are running at the same time, including the Seastar subsystem that is built into the Redpanda binary. Each of these processes consumes memory. You can configure the memory resources that are allocated to these processes.
```

**Review by RafalKorepta — APPROVED** [review]

File: docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-best-practices.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (joejulian) [reply docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-best-practices.mdx:166]: Existing text matches the chart expectations.

https://github.com/redpanda-data/helm-charts/blob/main/charts/redpanda/templates/_helpers.tpl#L186

File: docs/manage/kubernetes/manage-resources.mdx (hunk 3 above)
Comment (joejulian) [reply docs/manage/kubernetes/manage-resources.mdx:130]: The comment is incorrect. [This function](https://github.com/redpanda-data/helm-charts/blob/8acf816277f3a22d057491117d538ccca24a4bc0/charts/redpanda/templates/_helpers.tpl#L183-L214) will return bytes, regardless of the decimal places of the SI unit.

For memory units, we then convert that number to the nearest whole MiB

**Review by joejulian — APPROVED** [review]
Approving but hoping the "single digit numbers" line gets rephrased based on the new understanding.

**Pull request merged**